numpy/core/src/multiarray/hashdescr.c

```diff
@@ -104,18 +103,24 @@ static int _array_descr_builtin(PyArray_Descr* descr, PyObject *l)
  *
  * Return 0 on success
  */
-static int _array_descr_walk_fields(PyObject* fields, PyObject* l)
+static int _array_descr_walk_fields(PyObject *names, PyObject* fields, PyObject* l)
 {
-    PyObject *key, *value, *foffset, *fdescr;
+    PyObject *key, *value, *foffset, *fdescr, *ftitle;
     Py_ssize_t pos = 0;
     int st;
 
-    while (PyDict_Next(fields, &pos, &key, &value)) {
+    for (pos = 0; pos < PyTuple_GET_SIZE(names); pos++) {
```

Comment:
Shouldn't we explicitly check that `names` is indeed a tuple before calling `PyTuple_GET_SIZE` and `PyTuple_GET_ITEM` below?